packages/@dcl/inspector/src/components/tree/Tree.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react'
+import { useDrag, useDrop } from 'react-dnd'
+
+import './Tree.css'
+
+import { Tree, TreeType } from '../../utils/tree'
+import { Input } from './input'
+import { Controls } from './controls'
+import { Icon } from './icon'
+
+interface Props {
+  value: Tree
+}
+
+interface TreeProps {
+  tree: Tree
+  update: () => void
```

Comment:
Instead of an `update` callback that forces a re-render, we should comunicate the desired changes through callbacks, for example:

```ts
onRename: (id: string, newName: string) => void,
onSetParent: (id: string, newParent: string | null) => void
// and so on...
```

So then the parent can decide how to perform those changes and provide a new tree.